pkg/cloudclient/aws/aws.go

```diff
@@ -0,0 +1,212 @@
+package aws
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sts"
+
+	"github.com/golang/glog"
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	awsclient "github.com/openshift/configuration-anomaly-detection/pkg/cloudclient/aws/client"
+	ccerrors "github.com/openshift/configuration-anomaly-detection/pkg/cloudclient/errors"
+	"github.com/openshift/configuration-anomaly-detection/pkg/ocm"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/pkg/errors"
+)
+
+const (
+	jumpRole      = "ba"
+	defaultRegion = "us-east-1" // TODO extract to own config
+	credentialDir = "~/.aws"    // TODO extract to own config
+)
+
+const (
+	// the ClusterDeployment label key we're expecting to be able to pull the ClusterID from
+	cdClusterIDLabelKey string = "api.openshift.com/id"
+)
+
+// CloudClient is the client used for handling AWS Cloud Handler requests
+type CloudClient struct {
+	// clusterDeployment is the ClusterDeployment for the target cluster (the one
+	// for which a login url is requested)
+	clusterDeployment *hivev1.ClusterDeployment
+
+	// awsClient is the awsClient that has the BACKPLANE user credentials.  This client
+	// is used to create other clients in order to perform the necessary role chaining to
+	// access a customer's account via STS assume roles.
+	awsClient awsclient.AwsClient
+
+	// ocmClient is a client that can query OCM in order to get the AccountClaim
+	ocmClient ocm.Client
+
+	// awsClientBuilder is responsible for building the aws client in order to
+	// allow mocking during testing
+	awsClientBuilder awsclient.BuilderIface
```

Comment:
I'm not sure builderIface is the best name for this interface

Reply:
Yep, probably it's a ClientFactory or so is a better name.

Reply:
Also maybe the issue is the struct and the interface have a similar name

@rafael-azevedo had thoughts on where the interfaces should live